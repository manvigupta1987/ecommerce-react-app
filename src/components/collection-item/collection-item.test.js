import React from 'react';
import { shallow } from 'enzyme';
import { CollectionItem } from './collection-item.component.jsx';

describe('CollectionItem Component', ()=> {
  let wrapper;
  let mockAddItems;
  let imageUrl = 'testing'
  let name = 'testing'
  
  beforeEach(() => {
    mockAddItems = jest.fn()
    const mockProps = {
      item: {
        imageUrl: imageUrl,
        name: name,
        price: '$23'
      },
      addItems: mockAddItems 
    }
    wrapper = shallow(<CollectionItem {...mockProps}/>)
  })
  it('should render CollectionItem component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('ImageContainer should have imageUrl as prop', () => {
    expect(wrapper.find('ImageContainer').prop('imageUrl')).toBe(imageUrl);
  });

  it('NameContainer should have name title', () => {
    expect(wrapper.find('NameContainer').text()).toEqual(name);
  });

  it('CustomButtonContainer should call addItems ', () => {
    wrapper.find('CustomButtonContainer').simulate('click')
    expect(mockAddItems).toHaveBeenCalled();
  });
})

