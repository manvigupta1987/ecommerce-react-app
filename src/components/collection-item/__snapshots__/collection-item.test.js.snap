// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem Component should render CollectionItem component 1`] = `
<styled.div>
  <ImageContainer
    className="image"
    imageUrl="testing"
  />
  <styled.div>
    <NameContainer>
      testing
    </NameContainer>
    <styled.span>
      $
      $23
    </styled.span>
  </styled.div>
  <CustomButtonContainer
    inverted={true}
    onClick={[Function]}
  >
    Add to Cart
  </CustomButtonContainer>
</styled.div>
`;
